src/test/java/seedu/uninurse/model/tag/TagListTest.java

```diff
@@ -0,0 +1,155 @@
+package seedu.uninurse.model.tag;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.uninurse.testutil.Assert.assertThrows;
+import static seedu.uninurse.testutil.TypicalTags.TAG_ELDERLY;
+import static seedu.uninurse.testutil.TypicalTags.TAG_NURSING_HOME;
+import static seedu.uninurse.testutil.TypicalTags.TYPICAL_TAG_ELDERLY;
+import static seedu.uninurse.testutil.TypicalTags.TYPICAL_TAG_NURSING_HOME;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.uninurse.model.tag.exceptions.DuplicateTagException;
+import seedu.uninurse.model.tag.exceptions.TagNotFoundException;
+
+public class TagListTest {
+    private final TagList emptyTagList = new TagList();
+    private final TagList tagListElderly = new TagList(List.of(TAG_ELDERLY));
+    private final TagList tagList = new TagList(Arrays.asList(TAG_ELDERLY, TAG_NURSING_HOME));
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new TagList(null));
+    }
+
+    @Test
+    public void constructor_unsortedTagList_returnsSortedTagList() {
+        TagList tagList = new TagList(Arrays.asList(TAG_ELDERLY, TAG_NURSING_HOME));
+        TagList expectedTagList = new TagList(Arrays.asList(TAG_NURSING_HOME, TAG_ELDERLY));
+        assertEquals(tagList, expectedTagList);
+    }
+
+    @Test
+    public void add_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> emptyTagList.add(null));
+    }
+
+    @Test
+    public void add_duplicateTag_throwsDuplicateTagException() {
+        assertThrows(DuplicateTagException.class, () -> tagListElderly.add(TAG_ELDERLY));
+    }
+
+    @Test
+    public void add_validTag_success() {
+        TagList updatedTagList = emptyTagList.add(TAG_ELDERLY);
+        assertEquals(updatedTagList, tagListElderly);
+    }
```

Comment:
Maybe we can add an additional test case for `TagList` specifically to ensure newly added tags will still yield a sorted `TagList`?

Reply:
Ok I've updated for this suggestion. As for the others, we can discuss it during the meeting